dependencies/CMakeLists.txt

```diff
@@ -1,98 +1,120 @@
 # Initializes a git submodule if it hasn't been initialized before
 
-
-find_package(Git QUIET) # We want the library to build even if git is missing
-if ((Git_FOUND) AND SIMDJSON_GIT AND (SIMDJSON_IS_UNDER_GIT))
-  message(STATUS "Git is available.")
-  # Does NOT attempt to update or otherwise modify git submodules that are already initialized.
-  function(initialize_submodule DIRECTORY)
-    if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${DIRECTORY}/.git)
-      message(STATUS "${CMAKE_CURRENT_SOURCE_DIR}/${DIRECTORY}/.git does not exist. Initializing ${DIRECTORY} submodule ...")
-      execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init ${CMAKE_CURRENT_SOURCE_DIR}/${DIRECTORY}
-                      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-                      RESULT_VARIABLE GIT_EXIT_CODE)
-      if(NOT GIT_EXIT_CODE EQUAL "0")
-        message(FATAL_ERROR "${GIT_EXECUTABLE} submodule update --init dependencies/${DIRECTORY} failed with exit code ${GIT_EXIT_CODE}, please checkout submodules")
-      endif()
-    endif()
-  endfunction(initialize_submodule)
-
-  if (SIMDJSON_GOOGLE_BENCHMARKS)
-    message (STATUS "'SIMDJSON_GOOGLE_BENCHMARKS' is requested, configuring..." )
-    option(BENCHMARK_ENABLE_TESTING OFF)
-    set(BENCHMARK_ENABLE_TESTING OFF)
-    option(BENCHMARK_ENABLE_INSTALL OFF)
-    set(BENCHMARK_ENABLE_INSTALL OFF)
-    initialize_submodule(benchmark)
-    add_subdirectory(benchmark)
+include(FetchContent)
+
+option(SIMDJSON_COMPETITION "Compile competitive benchmarks" ON)
+option(SIMDJSON_GOOGLE_BENCHMARKS "compile the Google Benchmark benchmarks" ON)
+
+function(set_off NAME)
+  set("${NAME}" OFF CACHE INTERNAL "")
+endfunction()
+
+# Note: the two functions below must only receive lowercase arguments for NAME
+macro(declare_dependency NAME GITHUB_REPO GIT_TAG)
+  message(STATUS "Declaring dependency ${NAME} (${GITHUB_REPO}@${GIT_TAG})")
+  FetchContent_Declare("${NAME}"
+          GIT_REPOSITORY "https://github.com/${GITHUB_REPO}.git"
+          GIT_TAG "${GIT_TAG}"
```

Comment:
CMake is always a wild, crazy ride to me, so please pardon me if I've misunderstand how FetchContent is working.

I think you could speed this up by using `GIT_SHALLOW ON`, here. Some of these repos are quite large, and all we really want is the exact commit we're explicitly fetching. It seems like FetchContent is supposed to do this automatically, but only if the tag matches `v.*`. Docs are a little unclear.

Reply:
`GIT_SHALLOW` works only for actual tags, not commit hashes.

Since `FetchContent` delegates to `ExternalProject`, things not covered in the docs of the former are in the docs of the latter.  
You can see it [here](https://cmake.org/cmake/help/latest/module/ExternalProject.html#command:externalproject_add), just have to scroll down to `Download Step Options` > `Git`.

However, it seems `FetchContent` incurs overhead more overhead than necessary, so maybe it is not worth using.